build/shade/_dotnet-restore.shade

```diff
@@ -7,9 +7,24 @@ restoreDir=''
     Optional. The directory in which to execute the dnu restore command.
 */}
 
+default KOREBUILD_DOTNET_RESTORE_INFER_RUNTIMES = '${ E("KOREBUILD_DOTNET_RESTORE_INFER_RUNTIMES") == "1" }'
 default currentDir = '${ Directory.GetCurrentDirectory() }'
 default restoreDir = '${ currentDir }'
 
-default restore_options=' ${ E("KOREBUILD_DOTNET_RESTORE_NO_INFER_RUNTIMES") == "1" ? "" : "--infer-runtimes" } ${ E("KOREBUILD_DOTNET_RESTORE_OPTIONS") }'
+@{
+    var projectFiles = Files.Include(Path.Combine(restoreDir, "**/project.json")).ToArray();
+    bool inferRids = false;
+    foreach (var projectFile in projectFiles)
+    {
+        inferRids |= File.ReadAllText(projectFile).Contains("\"netstandardapp1.5\"");
+        if (inferRids)
+        {
+            Log.Warn("Forcing infer-runtimes because of: " + projectFile);
+        }
+    }
+    KOREBUILD_DOTNET_RESTORE_INFER_RUNTIMES |= inferRids;
+}
+
+default restore_options=' ${ KOREBUILD_DOTNET_RESTORE_INFER_RUNTIMES ? "--infer-runtimes" : "" } ${ E("KOREBUILD_DOTNET_RESTORE_OPTIONS") }'
```

Comment:
You could use `inferRids` isntead. Alsomight need a space after `--infer-runtimes`